checktypes.json

```diff
@@ -0,0 +1,151 @@
+{
+    "checktypes": [
+        {
+            "name": "vulcan-gitleaks",
+            "description": "Detect hardcoded secrets using Gitleaks",
+            "image": "vulcansec/vulcan-gitleaks:edge",
+            "timeout": 600,
+            "options": {
+                "branch": "",
+                "depth": 1,
+                "excludedRules": []
+            },
+            "required_vars": [
+                "GITHUB_ENTERPRISE_ENDPOINT",
+                "GITHUB_ENTERPRISE_TOKEN"
+            ],
+            "assets": [
+                "GitRepository"
+            ]
+        },
+
```

Comment:
This `checktypes.json` is based on the one published along with [vulcan-local](https://github.com/adevinta/vulcan-local/blob/8ea50298b75b46fc0496c4ab9e20bf71bf756def/resources/checktypes.json) where both, `vulcan-gitleaks` and `vulcan-trivy`, are enabled in the [internal-static policy](https://github.com/adevinta/vulcan-local/blob/8ea50298b75b46fc0496c4ab9e20bf71bf756def/resources/internal-policies.yaml#L2-L9)).

We plan to define the set of defaults for Lava during the next sprint, which includes selecting the checktypes that should be enabled by default. 

That being said, doesn't vulcan-gitleaks detect more secret types? I remember that we had to migrate a JFrog rule from Gitleaks to Trivy.

Reply:
> We plan to define the set of defaults for Lava during the next sprint, which includes selecting the checktypes that should be enabled by default.

OK, perfect!

> That being said, doesn't vulcan-gitleaks detect more secret types? I remember that we had to migrate a JFrog rule from Gitleaks to Trivy.

At least we found a case detected by Gitleaks and not detected by Trivy, but it might happen the other way around or with other tools covering other areas. In order to reduce the number of checks to maintain and to prevent duplicate findings, we conducted a check consolidation process and Trivy was the tool selected for secret analysis too.

Reply:
> In order to reduce the number of checks to maintain and to prevent duplicate findings, we conducted a check consolidation process and Trivy was the tool selected for secret analysis too.

Great! If you don't mind, I'll schedule a quick VC to sync about this once I catch up with the current existing checks.